docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,90 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
```

Comment:
We should probably state which operations are supported by the indices.

IIUC:

 * BTREE: `=`, `<>`, inequalities (`<=`, `>`, and so on)
 * BITMAP: `=`
 * LABEL_LIST: `array_has_any`, `array_has_all`, `array_contains`

Reply:
Added a table.